File: src/main/java/com/ptah/common/BaseDto.java
package com.ptah.common;

import cn.hutool.core.bean.BeanUtil;

public abstract class BaseDto<T> {

    public BaseDto() {
    }

    public T fromEntity(BaseEntity entity) {
        return (T) BeanUtil.toBean(entity, this.getClass());
    }

    public T toEntity(Class<T> entityClass) {
        return BeanUtil.toBean(this, entityClass);
    }
}

File: src/main/kotlin/com/ptah/common/BaseDto.kt
package com.ptah.common

import cn.hutool.core.bean.BeanUtil

abstract class BaseDto<T> {
    fun fromEntity(entity: BaseEntity?): T {
        return BeanUtil.toBean(entity, this.javaClass) as T
    }

    fun toEntity(entityClass: Class<T>?): T {
        return BeanUtil.toBean(this, entityClass)
    }
}
